### PR TITLE
[circle-operator] Introduce driver

### PR DESCRIPTION
**Pull request by seanshpark**
This will introduce circle-operator driver and empty dump run method.

ONE-DCO-1.0-Signed-off-by: SaeHie Park <saehie.park@gmail.com>

### CODE REVIEW

**Comment by seanshpark** [comment]
for #9163, from draft #9193

File: compiler/circle-operator/src/Dump.cpp [hunk 1]
```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright (c) 2019 Samsung Electronics Co., Ltd. All Rights Reserved
```
Comment (mhs4670go) [inline compiler/circle-operator/src/Dump.cpp:2]:
```suggestion
 * Copyright (c) 2022 Samsung Electronics Co., Ltd. All Rights Reserved
```

File: compiler/circle-operator/src/Dump.h [hunk 2]
```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright (c) 2019 Samsung Electronics Co., Ltd. All Rights Reserved
```
Comment (mhs4670go) [inline compiler/circle-operator/src/Dump.h:2]: ditto

File: compiler/circle-operator/src/Dump.h [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef __DUMP_H__
+#define __DUMP_H__
```
Comment (mhs4670go) [inline compiler/circle-operator/src/Dump.h:18]: Maybe?
```suggestion
#ifndef __CIROPS_DUMP_H__
#define __CIROPS_DUMP_H__
```

File: compiler/circle-operator/driver/Driver.cpp [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+    oStream = &oFstream;
+  }
+
+  // hook segment fault
+  struct sigaction sa;
+  memset(&sa, 0, sizeof(struct sigaction));
+  sigemptyset(&sa.sa_mask);
+  sa.sa_sigaction = handle_segfault;
+  sa.sa_flags = SA_SIGINFO;
+  sigaction(SIGSEGV, &sa, NULL);
```
Comment (mhs4670go) [inline compiler/circle-operator/driver/Driver.cpp:88]: What does these codes do? I mean when this segment fault happens?

File: compiler/circle-operator/src/Dump.h (hunk 3 above)
Comment (seanshpark) [reply compiler/circle-operator/src/Dump.h:18]: it's executable so uniqueness is not needed.

File: compiler/circle-operator/driver/Driver.cpp [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+  catch (const std::runtime_error &err)
+  {
+    std::cerr << err.what() << std::endl;
+    std::cerr << arser;
+    return 255;
+  }
+
+  cirops::DumpOption option;
+  option.names = !!arser["--name"];
+  option.codes = !!arser["--code"];
```
[the remark this answers is not in the document]
Comment (seanshpark) [reply compiler/circle-operator/driver/Driver.cpp:61]: for double NOT, I wasn't sure we just use `option.names = arser["--name"];`.

File: compiler/circle-operator/driver/Driver.cpp (hunk 4 above)
Comment (seanshpark) [reply compiler/circle-operator/driver/Driver.cpp:88]: Yup. `handle_segfault()` will be called. This is to not show `Segmentation fault` message by the system.

File: compiler/circle-operator/driver/Driver.cpp (hunk 5 above)
[the remark this answers is not in the document]
Comment (seanshpark) [reply compiler/circle-operator/driver/Driver.cpp:61]: we can :) `bool operator[](const std::string &arg_name)`

File: compiler/circle-operator/src/Dump.h [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+#include <ostream>
+
+namespace cirops
+{
+
+struct DumpOption
+{
+  bool names = false;
+  bool codes = false;
+  bool all_graphs = false;
```
[the remark this answers is not in the document]
Comment (seanshpark) [reply compiler/circle-operator/src/Dump.h:31]: Why you ask?

[the remark this answers is not in the document]
Comment (mhs4670go) [reply compiler/circle-operator/src/Dump.h:31]: I meant to say it wasn't needed. But looks optional:)

**Review by mhs4670go — APPROVED** [review]
LGTM

File: compiler/circle-operator/src/Dump.h (hunk 6 above)
[the remark this answers is not in the document]
Comment (seanshpark) [reply compiler/circle-operator/src/Dump.h:31]: Ah, just to be sure as `all_graphs` is used but not set not cleared.

**Pull request merged**